### PR TITLE
add extension nft-search

### PR DESCRIPTION
**Pull request by glazec**
## Description

Add new extension nft-search. Input collection name and it will return price and relative websites.

## Screencast
nft_search_1
nft_serach_2

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @glazec 👋 

Thanks for your contribution, could you start fix [this error](https://github.com/raycast/extensions/runs/6978571559?check_suite_focus=true) so I can start the review?

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/nft-search/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                url={"https://www.twitter.com/" + searchResult.twitter}
+                icon="twitter.png"
+              />
+            ) : null}
+            {searchResult.discord ? (
+              <Action.OpenInBrowser title="Open Discord in Browser" url={searchResult.discord} icon="discord.png" />
+            ) : null}
+            {searchResult.ins ? (
+              <Action.OpenInBrowser
+                title="Open Ins in Browser"
```
Comment (pernielsentikaer) [inline extensions/nft-search/src/index.tsx:55]: Shouldn't this have the same naming as the other in the list?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @glazec 👋 

Thanks for the update 💪 Much appreciated 💯 

* There might be something wrong with the timing, when searching for `@trending` it ends up with this

CleanShot 2022-06-27 at 12 30 44@2x

* Could you move images used in the `README.md` to a folder called `media` instead

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by glazec** [comment]
> Hi @glazec 👋
> 
> Thanks for the update 💪 Much appreciated 💯
> 
> * There might be something wrong with the timing, when searching for `@trending` it ends up with this
> 
> CleanShot 2022-06-27 at 12 30 44@2x
> 
> * Could you move images used in the `README.md` to a folder called `media` instead
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

I checked the @trending problem. I think this problem comes from searching timing. You are typing @trending, but the error says cannot find @tren. What happens is that raycast searches @tren when you are typing. Obviously no collection is called @tren. Thus you got an error. **I am not sure how we can improve this. Do you have any ideas?**

Plus I have moved the image to the media folder.

File: extensions/nft-search/src/index.tsx (hunk 1 above)
Comment (glazec) [reply extensions/nft-search/src/index.tsx:55]: For this problem, you mean naming convention? `serachResult.instagram` or `title="Open Instagram in Browser"`

Comment (pernielsentikaer) [reply extensions/nft-search/src/index.tsx:55]: The title, since the other ones don't have?

Comment (glazec) [reply extensions/nft-search/src/index.tsx]: Resolved

**Comment by glazec** [comment]
@pernielsentikaer Hi, any update for this new extension?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @glazec 👋 

Sorry for the late reply; I was three weeks on vacation. I'm ready to get this finish now 😄 

Regarding the **@trending** issue, it seems to happen when I'm typing slowly. Some ways around it could be:

- Using a dropdown filter instead for trending entries (All | Trending)
- Use your own throttle and control the throttle duration

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by glazec** [comment]
I test in latest version of Raycast, I believe this issue is mitigated. Could you test again?

**Comment by pernielsentikaer** [comment]
I still have the same issue @glazec 😞

**Comment by glazec** [comment]
I will remove @trending in the search function at this time. I think is not elegant to use the filter for this.

File: extensions/nft-search/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# NFT Searching
+
+This project is developed by Glaze from un.Block. **We are trying to improve the crypto UX for Raycast community.** Check out our Twitter account [@unblock256](https://www.twitter.com/unblock256) and [@unblcok256_en](https://twitter.com/unblock256_en) for more information.
+
+For crypto community, check out our [guide](raycast.unblock256.com) for crypto kit including necessary crypto plugin for raycast users.
```
Comment (pernielsentikaer) [inline extensions/nft-search/README.md:5]: Link doesn't seem to be public?

File: extensions/nft-search/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+![nft_search_1](media/nft_search_1.png)
+![nft_search_2](media/nft_search_2.png)
```
Comment (pernielsentikaer) [inline extensions/nft-search/README.md:7]: Maybe just use the new metadata image instead

File: extensions/nft-search/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,205 @@
+import { Icon, ActionPanel, Action, List, showToast, Toast } from "@raycast/api";
+import { useState, useEffect, useRef, useCallback } from "react";
+import fetch, { AbortError, Headers } from "node-fetch";
+
+export default function Command() {
+  const { state, search } = useSearch();
+  return (
+    <List isLoading={state.isLoading} onSearchTextChange={search} searchBarPlaceholder="Ttype NFT name..." throttle>
```
Comment (pernielsentikaer) [inline extensions/nft-search/src/index.tsx:8]:
```suggestion
    <List isLoading={state.isLoading} onSearchTextChange={search} searchBarPlaceholder="Type NFT name..." throttle>
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @glazec 👋 

I made a new metadata image for you since one of them is not possible anymore and the other one didn't have the right format.

You can use this image for metadata:
nft-search-1

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/nft-search/README.md (hunk 2 above)
Comment (glazec) [reply extensions/nft-search/README.md:5]: I updates the DNS configuration

**Comment by pernielsentikaer** [comment]
Could you also only use the image I shared as metadata? I can't edit this PR myself so I have to wait for you 🙂

**Comment by glazec** [comment]
Update the metadata

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by glazec** [comment]
On the Readme, the guide link is correct. But on the website, it is https://github.com/raycast/extensions/blob/main/extensions/nft-search/raycast.unblock256.com which is incorrect.

**Comment by pernielsentikaer** [comment]
Should be better now @glazec, there was no `https://` on the link

**Pull request merged**